mail/src/main/scala/spinoco/protocol/mail/header/codec/RFC2047Codec.scala

```diff
@@ -80,29 +80,35 @@ object RFC2047Codec {
       if (encode.forall { c => AsciiEncoder.canEncode(c)  && ! c.isControl && c != '?' }) Attempt.successful(encode)
       else {
         @tailrec
-        def go(remains: String, buff: String, acc: String): Attempt[String] = {
+        def go(remains: String, buff: String, acc: String, highSurrogate: Option[Char]): Attempt[String] = {
           remains.headOption match {
             case Some(ch) =>
-              val encoded: String = ch match {
-                case ' ' => "_"
-                case '_' => "=5F"
-                case '?' => "=3F"
-                case '=' => "=3D"
+              val encoded: Option[String] = ch match {
+                case ' ' => Some("_")
+                case '_' => Some("=5F")
+                case '?' => Some("=3F")
+                case '=' => Some("=3D")
                 case c =>
-                  if (AsciiEncoder.canEncode(c) && !c.isControl && c != '?') c.toString
-                  else ByteVector.view(UTF8Encoder.encode(CharBuffer.wrap(Array(c)))).toHex.toUpperCase.grouped(2).flatMap { "=" + _ }.mkString
+                  if (c.isHighSurrogate) None
+                  else if (AsciiEncoder.canEncode(c) && !c.isControl && c != '?') Some(c.toString)
```

Comment:
Do we need to check the question mark here? we already captured the case of "?" in the pattern match.